### PR TITLE
Propagate tests

### PR DESCRIPTION
**Pull request by luminoso-beaudoin**
Adds a new file to the small build tests that runs tests of the propagation code.  Also includes minor changes to the code under test to fix some corner cases that show up in testing (but would be unlikely to arise in practice).  Changed the name of variables in propagation and retrofitting code (to improve clarity).

### CODE REVIEW

File: tests/small-build/test_propagate.py [hunk 1]
```diff
@@ -0,0 +1,551 @@
...
+    # Call the code under test.
+    propagated = propagate(
+        COMBINED_INDEX, FRAME, ADJACENCY_MATRIX, len(NEW_ENGLISH_TERMS)
+    )
+
+    # The propagated terms should be the terms from the conbined index,
+    # starting with the terms of the frame and going up to the last new
+    # term from the graph that is not in English.
+    assert (len(propagated) == len(FRAME) + len(NEW_NON_ENGLISH_TERMS)), \
+        'Incorrect number {} (should be {}) of propataged terms.'.format(len(propagated), len(FRAME) + len(NEW_NON_ENGLISH_TERMS))
```
Comment (rspeer) [inline tests/small-build/test_propagate.py]: typo: "propataged"
